### DISCUSSION TITLE
UART RX not working on B-L072Z-LRWAN1

### DISCUSSION THREAD

**Discussion by josesimoes** [Lora-net/LoRaMac-node#1217] [Q&A]
Hi,

The UART RX is not working only the 1st char is received.

Suspect this is because of the way the UART interrupt is being handled...
The repeated call to `HAL_UART_Receive_IT` inside `HAL_UART_RxCpltCallback` is doomed to fail because of the way the ST UART HAL code works. There are various entries about this on ST forums and Stackoverflow too.

Can this be reworked to implement a workaround for those shortcomings? Or maybe switching to DMA...

**Comment by zyk6271** [comment]
I meet this questions yesterday , I have solved it , let me tell you . In uart-board.c with line 220 . The name of func is UartMcuGetChar . Modify it to what I show to you. Here are the code.
`
uint8_t UartMcuGetChar( Uart_t *obj, uint8_t *data )
{
			if( IsFifoEmpty( &obj->FifoRx ) == false )
			{
					*data = FifoPop( &obj->FifoRx );
					HAL_Delay(1);
					return 0;
			}
			else
			{
					return 1;	
			}
}
`

**Comment by mluis1** [comment]
Can we close this issue ?

**Comment by josesimoes** [comment]
@mluis1 I still can't receive on the UART...

**Comment by kevin-malter** [comment]
Hi @josesimoes did you resolved the problem?

**Comment by josesimoes** [comment]
Hi @kevenmalter the suggestion above seems to work. Shouldn't it be ported to the respective code file?

**Comment by mluis1** [comment]
If you don't mind we decided to move this issue to the [Discussion](https://github.com/Lora-net/LoRaMac-node/discussions) tab as we are not able to reproduce this issue using the provided examples.
